ibc-testkit/src/testapp/ibc/core/core_ctx.rs

```diff
@@ -298,46 +268,449 @@ impl ValidationContext for MockContext {
+                    .client_state_store
+                    .get(StoreHeight::Pending, &client_state_path)
+                    .ok_or_else(|| ClientError::ClientStateNotFound {
+                        client_id: client_state_path.0.clone(),
+                    })?;
+                Ok((client_state_path.0, client_state))
+            })
+            .collect()
+    }
+
+    /// Returns the list of all consensus states of the given client.
+    fn consensus_states(
+        &self,
+        client_id: &ClientId,
+    ) -> Result<Vec<(Height, ConsensusStateRef<Self>)>, ContextError> {
+        let path = format!("clients/{}/consensusStates", client_id)
+            .try_into()
+            .map_err(|_| ClientError::Other {
+                description: "Invalid consensus state path".into(),
+            })?;
+
+        self.consensus_state_store
+            .get_keys(&path)
+            .into_iter()
+            .filter_map(|path| {
+                if let Ok(Path::ClientConsensusState(consensus_path)) = path.try_into() {
+                    Some(consensus_path)
+                } else {
+                    None
+                }
+            })
+            .map(|consensus_path| {
+                let height = Height::new(
+                    consensus_path.revision_number,
+                    consensus_path.revision_height,
+                )?;
+                let client_state = self
+                    .consensus_state_store
+                    .get(StoreHeight::Pending, &consensus_path)
+                    .ok_or({
+                        ClientError::ConsensusStateNotFound {
+                            client_id: consensus_path.client_id,
+                            height,
+                        }
+                    })?;
+                Ok((height, client_state))
+            })
+            .collect()
+    }
+
+    /// Returns the list of heights at which the consensus state of the given client was updated.
+    fn consensus_state_heights(&self, client_id: &ClientId) -> Result<Vec<Height>, ContextError> {
+        let path = format!("clients/{}/consensusStates", client_id)
+            .try_into()
+            .map_err(|_| ClientError::Other {
+                description: "Invalid consensus state path".into(),
+            })?;
+
+        self.consensus_state_store
+            .get_keys(&path)
+            .into_iter()
+            .filter_map(|path| {
+                if let Ok(Path::ClientConsensusState(consensus_path)) = path.try_into() {
+                    Some(consensus_path)
+                } else {
+                    None
+                }
+            })
+            .map(|consensus_path| {
+                Ok(Height::new(
+                    consensus_path.revision_number,
+                    consensus_path.revision_height,
+                )?)
+            })
+            .collect::<Result<Vec<_>, _>>()
+    }
+
+    /// Connections queries all the IBC connections of a chain.
```

Comment:
What is being referred to by "Connections" in this sentence. It seems to indicate that there's a "Connections" object that is querying all the IBC connections of a chain, but I'm not sure what that's referring to.

Reply:
Hmm. I don't remember this - but it seems to refer the function names. Anyway, I updated them.

b8db190d